scrapy/linkextractors/htmlparser.py

```diff
@@ -39,7 +40,7 @@ def _extract_links(self, response_text, response_url, response_encoding):
         ret = []
         base_url = urljoin(response_url, self.base_url) if self.base_url else response_url
         for link in links:
-            if isinstance(link.url, unicode):
+            if isinstance(link.url, six.text_type):
```

Comment:
isn't `six.text_type` the native string? > "This is unicode() in Python 2 and str in Python 3."
Or did you mean bytes in Python 3 and unicode in Python 2?

I guess this should use `to_native_str()` here, then?